backend-app/controllers/users_controllers/admin_controller.ts

```diff
@@ -197,92 +137,117 @@ export const unbanUser = async (req: IReq, res: IRes, next: INext) => {
             throw new AppError(400, 'User is not banned');
         user.accessRestricted = false;
         await user.save();
-        res.status(200).json({
+        this.setStatus(200);
+        return {
             message: 'User is now unbanned',
-        });
-    } catch (err) {
-        next(err);
+        };
     }
-};
 
-export const createRole = async (req: IReq, res: IRes, next: INext) => {
-    const { name, authorities, restrictions } = req.body;
-    try {
+    @Response(400, 'Role already exists')
+    @SuccessResponse('201', 'CREATED')
+    @Middlewares(restrictTo(Actions.MANAGE_ROLES))
+    @Security('jwt')
+    @Post('role')
+    async createRole(
+        @Body()
+        body: RoleType
+    ): Promise<{ message: string; data: RoleType }> {
+        const { name, authorities, restrictions } = body;
         if (await Role.getRoleByName(name))
             throw new AppError(400, 'Role already exists');
         const createdRole = await Role.createRole(
             name,
             authorities,
             restrictions
         );
-        res.status(201).json({
+        this.setStatus(201);
```

Comment:
Can you check if this is working because, for me, I do this:
```ts
public async updateMe(
        @Request() req: IReq,
        @Res() res: TsoaResponse<200, any>
    ): Promise<any> {
     //........
     return res(204, { message: 'User update successfully' });
}
```
what do you think ?

Reply:
Both works.

Reply:
then it is fine but we have to choose one. By the way, btw what does 'this' refer to because it is not clear?
@muttaqin1

Reply:
In that methods context "This" is here referring to the controller class. No worries, i will go for your approach.

Reply:
This class has some properties called 
```js
Class someContoller{
constructor(statusCode){
this.statusCode= statusCode;
}

setStatus(statusCode){
this.statusCode= statusCode;
}

Login(){
this.setStatus(200);
return {}//some response data
}
/*so when this login method runs the status code is set to 200 for this method. The controller takes the current status code and send response with this status code. This **statusCode** will be modified for other methods.
*/
//this is just my assumption.
}
```

Reply:
@bellaabdelouahab

Reply:
but what about if the class has 100 request each one is changing the 'this' hhhhh ?

Reply:
You will be shocked to know that behind the scene   **res** also does the same.

Reply:
Oh, really, if that's the case, then we can keep both but only for code clarity 
@muttaqin1

Reply:
No, i have already told you that i am going to follow your approach. Your approach is very clear and the code looks well👌